### PR TITLE
Add config option for disabling email MX check

### PR DESCRIPTION
**Pull request by skgsergio**
This PR adds the option for disabling the MX check over the user email when a user tries to register.

This is required in some environments without Internet connectivity.

### CODE REVIEW

File: config/app_config.yml.sample [hunk 1]
```diff
@@ -764,7 +764,7 @@ defaults: &defaults
         isolines: false
         routing: false
         data_observatory: false
-
+  disable_email_mx_check: # true / false (default). Disables MX DNS record check and just performs syntax validation.
```
Comment (gonzaloriestra) [inline config/app_config.yml.sample]: Maybe it's easier to understand this way:
```suggestion
  disable_email_mx_check: false # Disables MX DNS record check and just performs syntax validation.
```

**Comment by skgsergio** [comment]
## How to CR
Since in staging we check the mail also in central the best way to check this is working is the following:

```
$ RAILS_ENV=staging BUILDER_ENABLED=true SUBDOMAIN=thisshouldwork PASSWORD=thisshouldwork12345 EMAIL=sconde+test@carto.com bundle exec rake cartodb:db:setup_user
User thisshouldwork created successfully

$ RAILS_ENV=staging BUILDER_ENABLED=true SUBDOMAIN=patatasfritas PASSWORD=patatasfritas12345 EMAIL=test@carto.lan bundle exec rake cartodb:db:setup_user
rake aborted!
{:email=>["Domain name not registered"]}

Tasks: TOP => cartodb:db:setup_user
(See full trace by running task with --trace)
```

Then add `disable_email_mx_check: false` to the config and restart both unicorn and resque and test again, you should get the same results:

```
$ RAILS_ENV=staging BUILDER_ENABLED=true SUBDOMAIN=thisshouldwork2 PASSWORD=thisshouldwork12345 EMAIL=sconde+test2@carto.com bundle exec rake cartodb:db:setup_user
User thisshouldwork2 created successfully

$ RAILS_ENV=staging BUILDER_ENABLED=true SUBDOMAIN=patatasfritas PASSWORD=patatasfritas12345 EMAIL=test@carto.lan bundle exec rake cartodb:db:setup_user
rake aborted!
{:email=>["Domain name not registered"]}

Tasks: TOP => cartodb:db:setup_user
(See full trace by running task with --trace)
```

And then change it to `disable_email_mx_check: true`:

```
$ RAILS_ENV=staging BUILDER_ENABLED=true SUBDOMAIN=thisshouldwork3 PASSWORD=thisshouldwork12345 EMAIL=sconde+test3@carto.com bundle exec rake cartodb:db:setup_user
User thisshouldwork3 created successfully

$ RAILS_ENV=staging BUILDER_ENABLED=true SUBDOMAIN=patatasfritas PASSWORD=patatasfritas12345 EMAIL=test@carto.lan bundle exec rake cartodb:db:setup_user
User patatasfritas created successfully
```

**Comment by gonzaloriestra** [comment]
:heavy_check_mark: Acceptance OK

@skgsergio sorry for the delay :see_no_evil:

**Comment by gonzaloriestra** [comment]
retest this please

**Comment by gonzaloriestra** [comment]
retest this please

**Comment by gonzaloriestra** [comment]
retest this please

**Comment by gonzaloriestra** [comment]
retest this please

**Pull request merged**